templates/contribute/commit.md

```diff
@@ -0,0 +1,95 @@
+# Pull request title and description conventions
+
+We are using the following convention for writing pull request titles and descriptions.
+
+## Format
+
+Note: "Title:" and "Description:" do not actually appear
+
+```markdown
+  Title:
+  <type>(<optional-scope>): <subject>
+
+  Description:
+  <body>
+  <NEWLINE>
+  <footer>
+  <NEWLINE>
+  <dependencies>
+```
+
+`<type>` is:
+
+ - feat (feature)
+ - fix (bug fix)
+ - doc (documentation)
+ - style (formatting, missing semicolons, ...)
+ - refactor
+ - test (when adding missing tests)
+ - chore (maintain)
+ - perf (performance improvement, optimization, ...)
+
+`<optional-scope>` is a name of module or a directory which contains changed modules.
+This is not necessary to include, but may be useful if the `<subject>` is insufficient.
+The `Mathlib` directory prefix is always omitted.
+For instance, it could be
+
+- Data/Nat/Basic
+- Algebra/Group/Defs
+- Topology/Constructions
+
+`<subject>` has the following constraints:
+
+- use imperative, present tense: "change" not "changed" nor "changes"
+- do not capitalize the first letter
+- no dot(.) at the end
+
+`<body>` has the following constraints:
+
+- just as in ``<subject>``, use imperative, present tense
+- include motivation for the change and contrast with previous
+  behavior
+
+`<footer>` is optional and may contain two items:
+
+- Breaking changes: All breaking changes have to be mentioned in
+  footer with the description of the change, justification and
+  migration notes
```

Comment:
Do we actually care about this? I don't think I've seen this used in a commit message for mathlib before

Reply:
I think we actually do, although we're bad about enforcing it. There was talk in some Zulip thread which I can't find at the moment about doing exactly this so we can easily find it later. I think there was even discussion about the exact format it should take, but I don't think it was finalized.